egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
   this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
 
+  if [ "$adaptive_shrink" == "true" ]; then
+    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
```

Comment:
That graph could be interpreted as telling us that 0.95 shrinkage was too
high at the end, but 0.95, 0.98 and 0.99 were all in a reasonable range at
the beginning.  So it could be also be interpreted as saying that it's
important to aggressively decrease the shrinkage towards the end.  Anyway,
up to you.

On Sat, Sep 26, 2015 at 1:13 AM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/wsj/s5/steps/nnet3/lstm/train.sh
> <https://github.com/kaldi-asr/kaldi/pull/180#discussion_r40490302>:
>
> > @@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
> >    ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
> >    this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
> >
> > +  if [ "$adaptive_shrink" == "true" ]; then
> > +    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
>
> I take back my statement that initial shrink value is more important than
> rate.. I was looking at the results for various shrink values (See below
> plot) I think rate would be more important. So I implementing your formula..
>
> [image: shrink_vs_cvaccuracy]
> <https://cloud.githubusercontent.com/assets/3895456/10115967/97958f4e-63eb-11e5-8469-81968c06ee70.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/180/files#r40490302>.
>